src/plone/formwidget/recaptcha/norecaptcha.py

```diff
@@ -8,31 +8,113 @@
 
 import six
 
-
 try:
     import json
 except ImportError:
     import simplejson as json
 
-
 VERIFY_SERVER = "www.google.com"
 
 
+logger = logging.getLogger("plone.formwidget.recaptcha.norecaptcha")
+
+
 class RecaptchaResponse(object):
-    def __init__(self, is_valid, error_code=None):
+    def __init__(self, is_valid, error_code=None, score=None, action=None):
         self.is_valid = is_valid
         self.error_code = error_code
+        self.score = score
+        self.action = action
 
     def __repr__(self):
-        return "Recaptcha response: {0} {1}".format(self.is_valid, self.error_code)
+        return "Recaptcha response: {0} {1} score={2} action={3}".format(
+            self.is_valid, self.error_code, self.score, self.action
+        )
 
     def __str__(self):
         return self.__repr__()
 
 
-def displayhtml(
-    site_key, language="", theme="light", fallback=False, d_type="image", size="normal"
-):
+def _make_verification_request(data, verify_server=VERIFY_SERVER):
+    """
+    Common function to make verification requests to Google's reCAPTCHA API.
+    """
+    params = parse.urlencode(data)
+    request = Request(
+        url="https://{0}/recaptcha/api/siteverify".format(verify_server),
+        data=params,
+        headers={
+            "Content-type": "application/x-www-form-urlencoded",
+            "User-agent": "noReCAPTCHA Python",
+        },
+    )
+    if six.PY3:
+        request.data = request.data.encode("utf-8")
+
+    httpresp = urlopen(request)
+    return_values = json.loads(httpresp.read())
+    httpresp.close()
+
+    return return_values
+
+
+def displayhtml_v3(site_key, action="homepage"):
+    """
+    Returns HTML/JS for reCAPTCHA v3 integration.
+    site_key -- The site key
+    action -- The action name for v3 (e.g., 'homepage', 'login', etc.)
+    """
+    return '''
+<script src="https://www.google.com/recaptcha/api.js?render={SiteKey}"></script>
+<script>
+grecaptcha.ready(function() {{
+    grecaptcha.execute('{SiteKey}', {{action: '{Action}'}}).then(function(token) {{
+        var recaptchaResponse = document.getElementById('g-recaptcha-response');
+        if (recaptchaResponse) {{
+            recaptchaResponse.value = token;
+        }}
+    }});
+}});
+</script>
+<input type="hidden" id="g-recaptcha-response" name="g-recaptcha-response">
+'''.format(SiteKey=site_key, Action=action)
+
+
+def submit_v3(recaptcha_response_field, secret_key, remoteip=None, action=None, min_score=0.5, verify_server=VERIFY_SERVER):
+    """
+    Verifies reCAPTCHA v3 token.
+    recaptcha_response_field -- The token from the client
```

Comment:
+1 for the docstring. But it would be better if you would write it in the Plone styleguide suggested format:
https://5.docs.plone.org/develop/styleguide/python.html#docstrings-style